src/Foundation/NSDate.cs

```diff
@@ -65,42 +74,20 @@ public partial class NSDate {
 
 			var dtUnv = dt.ToUniversalTime ();
 
-			var calendar = new NSCalendar (NSCalendarType.Gregorian);
-			calendar.TimeZone = NSTimeZone.FromName ("UTC");
-			var components = new NSDateComponents {
-				Day = dtUnv.Day, Month = dtUnv.Month, Year = dtUnv.Year, Hour = dtUnv.Hour,
-					Minute = dtUnv.Minute, Second = dtUnv.Second, Nanosecond = dtUnv.Millisecond * 1000000
-			};
+			components.Day = dtUnv.Day;
+			components.Month = dtUnv.Month;
+			components.Year = dtUnv.Year;
+			components.Hour = dtUnv.Hour;
+			components.Minute = dtUnv.Minute;
+			components.Second = dtUnv.Second;
+			components.Nanosecond = Convert.ToInt32 (dtUnv.Millisecond * NANOSECS_PER_MILLISEC);
```

Comment:
Why do we need to use Convert here?

Reply:
`NANOSECS_PER_MILLISEC` is a double, so the result of the multiplication is a double...the compiler was throwing an error because `components.Nanosecond` is an int. The constant needs to store decimals for correct conversion/rounding in the other operator method.

Any suggestions to make this less clunky are very welcome :)

Reply:
```suggestion
			components.Nanosecond = (int)(dtUnv.Millisecond * NANOSECS_PER_MILLISEC);
```

Should work just fine.

Reply:
NANOSECS_PER_MILLISEC doesn't have to be a double, you can make it an int and that would avoid the cast as well.

Reply:
@rolfbjarne I believe that we need it as a `double` to avoid truncation (and correctly round) in `Convert.ToInt32 (calComponents.Nanosecond / NANOSECS_PER_MILLISEC)`. I could (very likely) be wrong here though 😆

Reply:
@whitneyschmidt there's one way to find out: write tests :smile: Try making `NANOSEC_PER_MILLISEC` an int, and then try to come up with a test that breaks the logic.

Reply:
When we're converting from ns to ms `Convert.ToInt32 (calComponents.Nanosecond / NANOSECS_PER_MILLISEC)` the constant needs to be a double so that we can round correctly in the conversion to ms.

I tried to reuse the constant again...but it does require the cast to `int` that wouldn't be necessary if `1000000` or an int constant was used instead.

Reply:
@rolfbjarne Tests confirm: we need a double, otherwise conversions truncate, rather than rounding 😄

Reply:
@whitneyschmidt that's great, please add those tests to the PR!